caddyhttp/httpserver/context.go

```diff
@@ -26,6 +26,7 @@ import (
 // Context is the context with which Caddy templates are executed.
 type Context struct {
 	Root http.FileSystem
+	Res  http.ResponseWriter
```

Comment:
I like your idea and we could get even better: don't export response's Header at all. This will forbid templates manipulating response header (e.g. Add, Set, Del). See my new patchset.